rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,311 @@
+            for token_address in token_addresses:
+                try:
+                    token = get_or_create_evm_token(
+                        userdb=evm_inquirer.database,
+                        evm_address=string_to_evm_address(token_address),
+                        chain_id=evm_inquirer.chain_id,
+                        protocol=CPT_GEARBOX,
+                        evm_inquirer=evm_inquirer,
+                        underlying_tokens=underlying_token,
+                        encounter=TokenEncounterInfo(
+                            description='Querying gearbox pools',
+                            should_notify=False,
+                        ),
+                    )
+                    verified_tokens[token_type].add(token.identifier)
+                except NotERC20Conformant as e:
+                    log.error(
+                        f'Skipping pool {pool} because {token_address} is not a '
+                        f'valid ERC20 token. {e}',
+                    )
+                    continue
+
+        pool.farming_pool_token = verified_tokens['farm'].pop()
+        pool.lp_tokens = verified_tokens['pools']
+        verified_pools.append(pool)
+    return verified_pools
+
+
+def get_zapper_tokens(pool_data: list[str], underlying_token_address: ChecksumEvmAddress) -> set[ChecksumEvmAddress]:  # noqa: E501
+    """Extracts zapper tokens from the pool data, excluding the underlying token address
+    and the special ETH address."""
+    zapper_tokens = {
+        de_addr
+        for data in pool_data[23]  # ZapperInfo[]
+        for addr in data[1:]  # ignore zapper address, only consider tokenIn and tokenOut
+        if (de_addr := deserialize_evm_address(addr)) not in {underlying_token_address, ETH_SPECIAL_ADDRESS}  # noqa: E501
+    }
+    return zapper_tokens
+
+
+def query_gearbox_data_from_chain(
+        evm_inquirer: 'EvmNodeInquirer',
+        existing_pools: set[ChecksumEvmAddress],
+) -> list[GearboxPoolData] | None:
+    """
+    Query all Gearbox information(lp tokens, pools, lp coins) from data compressor.
+
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    data_compressor = evm_inquirer.contracts.contract(GEARBOX_DATA_COMPRESSOR)
+    pools_data = data_compressor.call(
+        node_inquirer=evm_inquirer,
+        method_name='getPoolsV3List',
+    )
+    gearbox_farming_pool_abi = evm_inquirer.contracts.abi('GEARBOX_FARMING_POOL')
+    new_pools = []
+    for pool_data in pools_data:
+        try:
+            pool_address = deserialize_evm_address(pool_data[0])
+        except DeserializationError:
+            log.error(f'Could not deserialize evm address {pool_data[0]} while decoding gearbox pool information')  # noqa: E501
+            continue
+
+        if pool_address in existing_pools:
+            continue
+
+        try:
+            pool_name: str = pool_data[4]
```

Comment:
this doesn't raise deserializationError

Reply:
I have added IndexError and coupled it with `deserialize_evm_address(pool_data[1])`